packaging/package.sh

```diff
@@ -0,0 +1,243 @@
+    case ${name} in
+        "Ubuntu")
+            installCmd='apt --no-install-recommends'
+            packages="${packages} rpm"
+            ;;
+        "CentOS Linux")
+            installCmd="yum"
+            packages="${packages} rpm-build"
+            ;;
+        *)
+            echo "Unknown or unsupported linux detected"
+            echo "Will skip 'Check and prepare build environment' step"
+            echo "Please, prepare you environment manually"
+            ;;
+    esac
+
+    # Install missing software if necessary
+    installFlag=false
+    if [[ ! -z "${installCmd}" ]]; then
+        for executable in ${executables}; do
+            command -v ${executable} &>/dev/null || {
+                installFlag=true
+                break
+            }
+        done
+        if ${installFlag}; then
+            echo "[INFO] Software installation required root privileges"
+            sudo ${installCmd} install ${packages}
+        fi
+    fi
+}
+
+
+# Prepare binary artifacts from sources
+prepBin () {
+    echo "Packaging from sources is not implemented"
+}
+
+
+# Check that binary archive exists and try to download it from Apache Dist Archive is not
+getBin () {
+    IGNITE_VERSION=$(cat rpm/apache-ignite.spec | grep Version: | head -1 | sed -r 's|.*:\s+(.*)|\1|')
+    BIN_NAME="apache-ignite-fabric-${IGNITE_VERSION}-bin.zip"
+    binPreparedFlag=false
+
+    if [ -f "${BIN_NAME}" ]; then
+    binPreparedFlag=true
+    fi
+    if ! ${binPreparedFlag}; then
+        curl -O https://archive.apache.org/dist/ignite/${IGNITE_VERSION}/${BIN_NAME}
+        binPreparedFlag=true
+    fi
+    if ! ${binPreparedFlag}; then
+        echo "[ERROR] Can't find | get Apache Ignite's binary archive"
+        exit 1
+    fi
+}
+
+
+# Build RPM package
+buildRPM () {
+    RPM_WORK_DIR="$(mktemp -d)"
+    rm -rfv *.rpm
+
+    # Prepare build layout
+    mkdir -pv ${RPM_WORK_DIR}/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+    cp -rfv ${BIN_NAME} rpm/{name.service,service.sh} ${RPM_WORK_DIR}/SOURCES
+    cp -rfv rpm/apache-ignite.spec ${RPM_WORK_DIR}/SPECS
+
+    # Assemble RPM packages
+    rpmbuild -bb -v --define "_topdir ${RPM_WORK_DIR}" ${RPM_WORK_DIR}/SPECS/apache-ignite.spec
+
+    # Gather RPMS
+    find ${RPM_WORK_DIR} -name "*.rpm" -exec mv -fv {} ${PACKAGING_DIR} \;
+}
+
+
+# Build DEB package
+buildDEB () {
+    DEB_WORK_DIR="$(mktemp -d)"
+    rm -rfv *.deb
+
+    # Check that RPMs for building exists
+    packageVersion="$(cat deb/changelog | head -1 | sed -r 's|.*\((.*)\).*|\1|')"
+    if [ ! -f "apache-ignite-${packageVersion}.noarch.rpm" ]; then
+        echo "[ERROR] RPM for converting to DEB not found"
+        exit 1
+    fi
+
+    # Unpack RPMs and prepare DEBs build layout
+    cd ${DEB_WORK_DIR}
+    cp -rfv ${PACKAGING_DIR}/apache-ignite-${packageVersion}.noarch.rpm ${DEB_WORK_DIR}
+    alien --scripts --verbose --keep-version --single apache-ignite-${packageVersion}.noarch.rpm
```

Comment:
There's support for running yum/apt as sudo but no support for running alien as root. And it's actually required.

Reply:
I recommend adding $SUDO which is empty if user is already root